quill-jasync-zio/src/main/scala/io/getquill/context/zio/ZConnection.scala

```diff
@@ -0,0 +1,72 @@
+package io.getquill.context.zio
+
+import com.github.jasync.sql.db.{ ConcreteConnection, QueryResult }
+import com.github.jasync.sql.db.pool.{ ConnectionPool => KConnectionPool }
+import zio.{ Has, RIO, Task, TaskManaged, ZIO, ZLayer, ZManaged, Tag }
+import scala.jdk.CollectionConverters._
+
+object ZConnection {
+
+  trait Service {
+    protected def takeConnection: TaskManaged[ConcreteConnection]
+
+    private[zio] final def transaction[A](action: RIO[Has[Service], A]): ZIO[Has[Service], Throwable, A] = {
+      //Taken from ConcreteConnectionBase.kt to avoid usage of pool.inTransaction
```

Comment:
Is this was `pool.inTransaction` actually does?

Reply:
yes. I took this logic from `ConcreteConnectionBase.kt` to avoid conversion